src/Options/Form.php

```diff
@@ -157,6 +157,8 @@ public function write(): void
 								} else if (file_exists($opt['path'])) {
 									/** @noinspection PhpIncludeInspection */
 									include $opt['path'];
+								} else if (!empty($opt['fn']) && !is_string($opt['fn']) && is_callable($opt['fn'])) {
+									$opt['fn']();
```

Comment:
Лучше сделать, чтобы функция генерировала html для вывода

```suggestion
									echo $opt['render']();
```

Reply:
Можно, но вывод может быть необязательным, что, конечно, маловероятно

И если функция будет выводить какой-то сложный контент, тогда придется использовать _ob_get_clean_
Поэтому, думаю, лучше просто оставить её вызов без _echo_

Reply:
Вывод из самой функции напрямую - это скорее сайд-эффект, чем ожидаемое поведение. Если сделать вывод возвращаемого значения через echo, для пользователя библиотеки API этой фичи будет удобнее. Внутри этой функции он сможет реализовать рендеринг, как ему будет угодно. Хоть рендерингом какого-нибудь своего шаблона, хоть сгенерирует HTML с помощью других утилит (типа nette/utils), хоть сам завернет свой вывод напрямую в буфферизацию.

В дальнейшем я планирую переделать архитектуру библиотеки. Разделить рендеринг самой формы настроек от кода, добавить нужные абстрации для описания настроек и пр. Поэтому для данной фичи так же необходимо определить четкое API, которое поможет в сохранении обратной совместимости и тестировании

Reply:
Прошло больше года, а обновлений так и не было
Хотя в планах, да, это есть

Reply:
Не так много свободного времени. На все что хотелось бы сделать не хватает

Reply:
Код, который изменили в этом комментарии, не может быть применен к этому PR?